riffdog/command_line.py

```diff
@@ -161,7 +161,17 @@ def main():
     results = rd._items  # FIXME - accessing internal object
 
     if parsed_args.json:
-        print(dumps(results, cls=ReportEncoder))
+
+        filtered_data = []
+        for item in results:
+            if not item.matched or parsed_args.show_matched:
+                filtered_data.append(item)
```

Comment:
Is line 165-168 related to this change?

Reply:
Good Catch! IT should be - let me just check the next couple of lines

Reply:
updated in next commit - these *should* be here, but the next lines should have been using the filtered_data not the original results.